lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -5433,6 +5434,11 @@ void ByteCodeGenerator::EmitPropLoad(Js::RegSlot lhsLocation, Symbol *sym, Ident
             case Js::PropertyIds::undefined:
                 opcode = Js::OpCode::LdUndef;
                 break;
+            case Js::PropertyIds::__chakraLibrary:
+                if (CONFIG_FLAG(LdChakraLib)) {
+                    opcode = Js::OpCode::LdChakraLib;
```

Comment:
Forgive any rambling as I'm not a JIT expert...
What if those JS Built-ins are inlined into the JIT? You'd need to ensure that the library has been loaded and is visible only to the built-in and that it doesn't leak. But you also want to make sure that it is loaded the minimum number of times (ideally 1). If it is still necessary to run a bytecode to get a reference to that library object, that's alright, just ensure that it is cached.